.github/workflows/test.yml

```diff
@@ -94,22 +107,54 @@ jobs:
         - zsh
         - dash
         - ksh
+        - mksh
+        - yash
+        - 'busybox sh'
+        - gash
+        - mrsh
+        #- 'toybox toysh'
+        #- oil
+        update_alternatives:
+        - y
+        - n
     steps:
     - name: Check out code
       uses: actions/checkout@v1
 
     - name: Test
-      run: |
-        sudo apt-get update
-        sudo apt-get install -y ${{ matrix.shell }}
-        ./ci/runTest.sh ${{ matrix.shell }}
+      run: sudo GETMICRO_REGISTER=${{ matrix.update_alternatives }} ./ci/installAltShell.sh ${{ matrix.shell }}
 
   test_alpine:
-    name: Test busybox on alpine
-    runs-on: ubuntu-20.04
+    name: Test busybox on alpine with ${{ matrix.update_alternatives }} update-alternatives
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        update_alternatives:
+        - y
+        - n
     steps:
     - name: Check out code
       uses: actions/checkout@v1
 
     - name: Test
+      env:
+        GETMICRO_REGISTER: ${{ matrix.update_alternatives }}
       run: ./ci/runDockerTest.sh alpine:latest sh
+
+  check_readme_checksum:
```

Comment:
The checksum gets automatically updated on publish (you can check out the publishing script here: https://github.com/benweissmann/getmic.ro/blob/master/publish.sh) -- I think it's important that it gets updated at publish time, and not when a change merges, so the README isn't outdated in between when the PR is merged and when I publish the new version. So I think this is a cool idea but not a good fit for the CI pipeline.

Reply:
This line of code really rubs me the wrong way, so I'm going to do a little fix up if you don't mind, ha ha.

```sh
OLDSHA=$(rg -o 'The sha256 checksum is `([a-f0-9]+)`' -r '$1' < README.md)
```